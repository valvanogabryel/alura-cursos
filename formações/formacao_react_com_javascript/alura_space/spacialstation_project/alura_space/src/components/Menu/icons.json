[
    {
        "name": "Início",
        "alt": "Ícone de uma casa",
        "path": "/assets/icons/home.png",
        "link": "./index.html",
        "id": 0
    },
    {
        "name": "Mais curtidos",
        "alt": "Ícone de um joinha",
        "path": "/assets/icons/mostLiked.png",
        "link": "./index.html",
        "id": 1
    },
    {
        "name": "Mais vistos",
        "alt": "Ícone de um olho",
        "path": "/assets/icons/mostViewed.png",
        "link": "./index.html",
        "id": 2
    },
    {
        "name": "Novos",
        "alt": "Ícone de uma estrela brilhante",
        "path": "/assets/icons/hot.png",
        "link": "./index.html",
        "id": 3
    },
    {
        "name": "Surpreenda-me",
        "alt": "Ícone de uma lâmpada",
        "path": "/assets/icons/surpriseMe.png",
        "link": "./index.html",
        "id": 4
    }
]