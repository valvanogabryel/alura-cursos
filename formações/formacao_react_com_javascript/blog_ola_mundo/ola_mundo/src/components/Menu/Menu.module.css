.navigation {
    height: 112px;
    display: flex;
    justify-content: center;
    align-items: center;
    column-gap: 3rem;
}

.link {
    font-size: 1.25rem;
    line-height: 1.5rem;
    color: var(--cor-fonte-principal);
    opacity: 0.8;
    transition: 0.3s;
    position: relative;
}

.link.active {
    opacity: 1;
}

.link.active::after {
    content: "";
    display: flex;
    width: 100%;
    height: 0.1rem;
    background: var(--cor-fonte-principal);
    margin: auto;
    box-sizing: border-box;
    animation: width 0.3s;
    position: absolute;
    top: 82%;
}

@keyframes width {
    from {
        width: 0%;
    }

    to {
        width: 100%;
    }
}

@media (max-width: 744px) {
    .navigation {
        height: 72px;
    }
}
