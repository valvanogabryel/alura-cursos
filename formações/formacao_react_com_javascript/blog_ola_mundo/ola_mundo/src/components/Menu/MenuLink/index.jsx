import React from 'react';
import { NavLink } from 'react-router-dom';

// O componente "Link" do react-router-dom funciona de maneira semelhante a tag âncora do HTML. Porém, sua particularidade é que, ao contrário da tag "a", o componente "Link" não recarrega a página.

// <a href="/pathname"></a>
// <Link to="/pathanme"></Link>

// O componente "NavLink" é um tipo especial de <Link> que sabe qual está "ativo". Útil para menus de navegação.

const MenuLink = ({ path, styles, children }) => {
    return (
        <NavLink
            className=
            {
                ({ isActive }) => `${styles.link} ${isActive ? styles.active : ''}`
            }
            to={path}
        >
            {children}
        </NavLink>
    );
}

export default MenuLink;