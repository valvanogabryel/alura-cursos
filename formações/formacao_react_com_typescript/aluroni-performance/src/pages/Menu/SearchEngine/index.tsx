import React, { useMemo } from 'react';
import { CgSearch } from 'react-icons/cg';
import styles from './SearchEngine.module.scss';

interface SearchEngineProps {
  search: string,
  setSearch: React.Dispatch<React.SetStateAction<string>>
}

const SearchEngine = ({ search, setSearch }: SearchEngineProps) => {
  const icon = useMemo(() =>
    <CgSearch
      size={20}
      color='#4c4d5e'
    />, []);

  return (
    <div className={styles.searchEngine}>
      <input
        type="text"
        value={search}
        placeholder='Buscar'
        onChange={(event) =>
          setSearch(event.target.value)}
      />
      {icon}
    </div>
  );
};

export default SearchEngine;