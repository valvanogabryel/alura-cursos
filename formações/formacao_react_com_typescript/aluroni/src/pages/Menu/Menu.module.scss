main {
    padding: 20px 250px;
}
