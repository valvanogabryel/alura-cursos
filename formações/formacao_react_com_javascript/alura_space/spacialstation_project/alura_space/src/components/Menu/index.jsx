import React from "react";
import icons from "./icons.json";

import styles from "./Menu.module.scss";

export const Menu = () => {
  return (
    <nav className={styles.menu}>
      <ul className={styles.menu__lista}>
        {icons.map((icon) => (
          <li key={icon.id} className={styles.menu__item}>
            <img src={icon.path} alt={icon.name} />
            <a href={icon.link}>{icon.name}</a>
          </li>
        ))}
      </ul>
    </nav>
  );
};
