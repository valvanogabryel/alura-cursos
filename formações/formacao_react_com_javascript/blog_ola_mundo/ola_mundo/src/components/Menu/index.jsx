import React from 'react';
import styles from './Menu.module.css';
import MenuLink from './MenuLink';

const Menu = () => {

    return (
        <header>
            <nav className={styles.navigation}>
                <MenuLink path='/' styles={styles} innerContent='InÃ­cio' />
                <MenuLink path='/aboutme' styles={styles} innerContent='Sobre mim' />
            </nav>
        </header>
    );
}

export default Menu;