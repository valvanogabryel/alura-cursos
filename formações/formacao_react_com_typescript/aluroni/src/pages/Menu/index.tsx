import { ReactComponent as Logo } from 'assets/logo.svg';
import { useState } from 'react';

import SearchEngine from './SearchEngine';
import Filters from './Filters';
import Ordenator from './Ordenator';
import Items from './Items';

import styles from './Menu.module.scss';

const Menu = () => {
    const [search, setSearch] = useState('');
    const [filter, setFilter] = useState<number | null>(null);
    const [ordenator, setOrdenator] = useState('');

    return (
        <main>
            <nav className={styles.nav}>
                <Logo />
            </nav>
            <header className={styles.header}>
                <div className={styles.header__text}>
                    A casa da massa!
                </div>
            </header>
            <section className={styles.menu}>
                <h3 className={styles.menu__title}>Card√°pio</h3>
                <SearchEngine
                    search={search}
                    setSearch={setSearch}
                />
                <div className={styles.menu__filters}>
                    <Filters
                        filter={filter}
                        setFilter={setFilter}
                    />
                    <Ordenator
                        ordenator={ordenator}
                        setOrdenator={setOrdenator}
                    />
                </div>
                <Items />
            </section>
        </main>
    );
}

export default Menu;