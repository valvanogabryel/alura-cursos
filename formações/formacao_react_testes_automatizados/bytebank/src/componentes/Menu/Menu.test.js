import { getAllByRole, render, screen } from "@testing-library/react";
import Menu from './index';

// ? Métodos de consulta
// !getBy:  
//* Este método é usado para selecionar um elemento com base em um seletor específico. Ele retorna o primeiro elemento que corresponde ao seletor fornecido e, se nenhum elemento for encontrado, ele lançará um erro. Este método é mais adequado para quando você espera que o elemento selecionado esteja presente no componente.

// !getAllBy:
//* Semelhante ao método getBy, este método é usado para selecionar vários elementos com base em um seletor. Ele retorna todos os elementos que correspondem ao seletor fornecido e, se nenhum elemento for encontrado, ele lançará um erro.

// !queryBy:
//* este método é semelhante ao método getBy, mas retorna null em vez de lançar um erro quando nenhum elemento é encontrado com base no seletor fornecido. Este método é mais adequado para quando você está procurando por um elemento que pode ou não estar presente no componente.

// !queryAllBy:
// * semelhante ao método queryBy, este método é usado para selecionar vários elementos com base em um seletor e retorna um array vazio se nenhum elemento for encontrado.

// !findBy:
// * este método é semelhante ao método getBy, mas é usado para selecionar elementos que podem não estar presentes no componente imediatamente após a renderização. Ele retorna um elemento que corresponde ao seletor fornecido e, se nenhum elemento for encontrado, ele aguarda até que um elemento correspondente seja adicionado ao DOM. Este método usa a API async/await e retorna uma Promise.

// !findAllBy:
// * semelhante ao método findBy, este método é usado para selecionar vários elementos que podem não estar presentes imediatamente após a renderização do componente. Ele retorna um array de elementos que correspondem ao seletor fornecido e aguarda até que pelo menos um elemento correspondente seja adicionado ao DOM. Este método também usa a API async/await e retorna uma Promise.



test('Should render the homepage link', () => {
  render(<Menu/>);
  const link = screen.getByText('Inicial');
  expect(link).toBeInTheDocument();
});

test('Should render a links list', () => {
  render(<Menu/>);
  const linksList = screen.getAllByRole('link');
  expect(linksList).toHaveLength(4);
});

test('Should not render a link for extract', () => {
  render(<Menu/>);
  const extractLink = screen.queryByText('Extrato');
  expect(extractLink).toBe(null);
  // ambos são iguais, mas manterei aqui para mostrar diferentes formas de fazer.
  expect(extractLink).not.toBeInTheDocument();
});

test('should render a list of links that contains the class name "link"', () => {
  render(<Menu/>);
  const links = screen.getAllByRole('link');
  links.forEach(link => expect(link).toHaveClass('link'));
  expect(links).toMatchSnapshot();
})