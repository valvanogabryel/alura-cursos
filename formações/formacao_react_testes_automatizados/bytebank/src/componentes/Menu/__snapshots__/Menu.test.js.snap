// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of links that contains the class name "link" 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
